internal/resources/syntheticmonitoring/resource_check_alert.go

```diff
@@ -0,0 +1,226 @@
+`,
+
+			CreateContext: withClient[schema.CreateContextFunc](resourceCheckAlertCreate),
+			ReadContext:   withClient[schema.ReadContextFunc](resourceCheckAlertRead),
+			UpdateContext: withClient[schema.UpdateContextFunc](resourceCheckAlertUpdate),
+			DeleteContext: withClient[schema.DeleteContextFunc](resourceCheckAlertDelete),
+
+			Importer: &schema.ResourceImporter{
+				StateContext: schema.ImportStatePassthroughContext,
+			},
+
+			Schema: map[string]*schema.Schema{
+				"check_id": {
+					Description: "The ID of the check to manage alerts for.",
+					Type:        schema.TypeInt,
+					Required:    true,
+					ForceNew:    true,
+				},
+				"alerts": {
+					Description: "List of alerts for the check.",
+					Type:        schema.TypeSet,
+					Required:    true,
+					ConfigMode:  schema.SchemaConfigModeAttr,
+					Elem: &schema.Resource{
+						Schema: map[string]*schema.Schema{
+							"name": {
+								Description: "Name of the alert. Must be one of: " + AlertNameProbeFailedExecutionsTooHigh + ", " + AlertNameTLSTargetCertificateCloseToExpiring,
+								Type:        schema.TypeString,
+								Required:    true,
+								ValidateFunc: validation.StringInSlice([]string{
+									AlertNameProbeFailedExecutionsTooHigh,
+									AlertNameTLSTargetCertificateCloseToExpiring,
+								}, false),
+							},
+							"threshold": {
+								Description: "Threshold value for the alert.",
+								Type:        schema.TypeFloat,
+								Required:    true,
+							},
+							"period": {
+								Description: "Period for the alert threshold. Required only for " + AlertNameProbeFailedExecutionsTooHigh + " alerts. One of: `1m`, `2m`, `5m`, `10m`, `15m`, `20m`, `30m`, `1h`.",
+								Type:        schema.TypeString,
+								Required:    false,
+								Optional:    true,
+								Default:     "",
+								ValidateFunc: validation.StringInSlice([]string{
+									"", "1m", "2m", "5m", "10m", "15m", "20m", "30m", "1h",
+								}, false),
+							},
+						},
+					},
+				},
+			},
+		},
+	)
+}
+
+func resourceCheckAlertCreate(ctx context.Context, d *schema.ResourceData, c *smapi.Client) diag.Diagnostics {
+	checkID := int64(d.Get("check_id").(int))
+
+	// Validate that the check exists
+	if err := validateCheckExists(ctx, c, checkID); err != nil {
+		return diag.FromErr(err)
+	}
+
+	alerts, err := makeCheckAlerts(d)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	_, err = c.UpdateCheckAlerts(ctx, checkID, alerts)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(fmt.Sprintf("%d", checkID))
+	return resourceCheckAlertRead(ctx, d, c)
+}
+
+func resourceCheckAlertRead(ctx context.Context, d *schema.ResourceData, c *smapi.Client) diag.Diagnostics {
+	checkID, err := strconv.ParseInt(d.Id(), 10, 64)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	// Validate that the check exists
+	if err := validateCheckExists(ctx, c, checkID); err != nil {
+		return diag.FromErr(err)
+	}
```

Comment:
Same comment as above. Since this is already done by the API, should rely on that and remove this from the client?